datalad/metadata/parsers/bids.py

```diff
@@ -102,27 +98,54 @@ def _get_dataset_metadata(self):
             meta['description'] = desc.strip()
 
         # special case
-        if bids.get('BIDSVersion'):
-            meta['conformsto'] = \
-                'http://bids.neuroimaging.io/bids_spec{}.pdf'.format(
-                    bids['BIDSVersion'].strip())
-        else:
-            meta['conformsto'] = 'http://bids.neuroimaging.io'
-        meta['@context'] = vocabulary
+        bids_version = meta.get('comment<BIDSVersion>', '').strip()
+        bids_defurl = 'http://bids.neuroimaging.io'
+        if bids_version:
+            bids_defurl += '/bids_spec{}.pdf'.format(bids_version)
+        meta['conformsto'] = bids_defurl
+        context['bids'] = {
+            # not really a working URL, but BIDS doesn't provide term defs in
+            # any accessible way
+            '@id': '{}#'.format(bids_defurl),
+            'description': 'ad-hoc vocabulary for the Brain Imaging Data Structure (BIDS) standard',
+            'type': vocabulary_id,
+        }
+        context.update(vocabulary)
+        meta['@context'] = context
         return meta
 
-    def _get_content_metadata(self):
+    def _get_cnmeta(self, bids):
+        # TODO any custom handling of participants infos should eventually
+        # be done by pybids in one way or another
+        path_props = {}
         participants_fname = opj(self.ds.path, 'participants.tsv')
         if exists(participants_fname):
             try:
-                for r in yield_participant_info(participants_fname):
-                    yield r
+                for rx, info in yield_participant_info(participants_fname):
+                    path_props[rx] = info
             except Exception as exc:
                 lgr.warning(
                     "Failed to load participants info due to: %s. Skipping the rest of file",
                     exc_str(exc)
                 )
 
+        # now go over all files in the dataset and query pybids for its take
+        # on each of them
+        for f in self.paths:
+            md = {}
+            try:
+                md.update(
+                    {'bids:{}'.format(k): v
+                     for k, v in bids.get_metadata(opj(self.ds.path, f)).items()})
+            except ValueError:
+                lgr.debug('no BIDS metadata for %s in %s', f, self.ds)
```

Comment:
please include actual exception into the message -- `exc_str` would be of help.  

Also it feels like there could be many reason BIDSLayout could emit some ValueError, which wouldn't really mean that there is no BIDS metadata there.

Reply:
Sadly, pybids is not very specific wrt to this error.